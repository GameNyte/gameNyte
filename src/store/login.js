const initialState = {
  userInfo: '',
  
}


export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {


    case 'CREATE_ACCOUNT':
      return { ...state, userInfo: payload.username};

    case 'LOGIN':
      return {...state, userInfo: payload.username };
    
   
      default:
      return state;
  }
}

export const createAccount = (userInfo) => {
  return {
    type: 'CREATE_ACCOUNT',
    payload: userInfo,
  }
}

export const login = (username, password) => {
  const userInfo = { 'username': username, 'password': password };
  return {
    type: 'LOGIN',
    payload: userInfo,
  }
}

