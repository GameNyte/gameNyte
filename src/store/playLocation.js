const initialState = {
  activeDrags: 0,
  deltaPosition: {
      x: 0, y: 0
  },
  controlledPosition: {
      x: -400, y: 200
  },
  socketId: null,
}  