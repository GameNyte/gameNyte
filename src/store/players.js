const initialState = [];



export default (state = initialState, action) => {
  const { type, payload } = action;  


  switch (type) {

    case 'ADD':
      return payload;

    case 'QUIT':
      return initialState;

    case 'UPDATE_LOCATION':      
    
      return state.map((player, idx) => {
        if(player.name === payload.name)  {
          return payload 
        }
        return player
      });


    default:
      return state;
  }
}

export const createPlayers = (player) => {
  return {
    type: 'ADD',
    payload: player,
  }
}



export const leavePlayers = (player) => {
  return {
    type: 'QUIT',
    payload: player,
  }
}


export const updateTokenLocation = (player) => {
  return {
    type: 'UPDATE_LOCATION',
    payload: player, 
  }
}
