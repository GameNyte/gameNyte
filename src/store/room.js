
const initialState = {
  room: '',
}


export default (state = initialState, action) => {
  const { type, payload } = action;


  switch (type) {


    case 'CREATE':
      return {...state, room: payload};

    case 'JOIN':
      return {...state, room: payload};

    case 'LEAVE':
      return initialState;  

    default:
      return state;
  }
}

export const createRoom = (room) => {
  return {
    type: 'CREATE',
    payload: room,
  }
}

export const joinRoom = (room) => {
  return {
    type: 'JOIN',
    payload: room,
  }
}

export const leaveRoom = () => {
  return {
    type: 'LEAVE',
  }
}