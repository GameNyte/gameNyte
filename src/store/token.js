
// const { default: players } = require("./players");

//needs to be a copy of all the players as well as x amount of extra players based on how many tokens were created ... 

const tokens = [];
// , ...extraTokens
const extraTokens = [];


const initialState = {
  tokens: [],
  extraTokens: [],
}

export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {

    case 'CREATE_TOKENS':
      return { ...state, tokens: [...payload] };

    case 'ADD_TOKEN':
      return { ...state, extraTokens: [...payload], tokens: [...tokens, ...extraTokens] };

    case 'RESET_TOKENS':
      return initialState;

    case 'UPDATE_LOCATION':
      return { ...state, } // need to update ALL tokens.player.location - the entire location

    default:
      return state;
  }
}

export const createTokens = (players) => {
  return {
    type: 'CREATE_TOKENS',
    payload: players,
  }
}

export const addToken = () => {

}

export const resetTokens = () => {

}

export const updateTokenLocation = (players) => {
  return {
    type: 'UPDATE_LOCATION',
    payload: players,
  }
}