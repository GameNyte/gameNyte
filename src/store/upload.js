const initialState = [];

export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {

    case 'UPDATE_MAPS':
      return [...state, payload];

    default:
      return state;
  }
}

export const handleAwsRes = (filename) => {
  return {
    type: 'UPDATE_MAPS',
    payload: filename,
  }
}
