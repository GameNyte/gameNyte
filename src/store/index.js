import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import room from './room.js';
import login from './login.js';
import players from './players.js';
import upload from './upload.js';

let reducers = combineReducers({ room, login, players, upload });

let store = () => {
  return createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));
};

export default store();