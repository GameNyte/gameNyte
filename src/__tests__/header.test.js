import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter });

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import { Provider } from 'react-redux';
import store from '../store';

import Header from '../components/header.js'

describe('Testing the Header', () => {
  it('Header should render at application start', () => {
    let app = shallow(<Header />);

    expect(app.find('div').exists()).toBe(true);
  });

  it('renders correctly', () => {
    const DOM = renderer.create(<Provider store={ store }><Header /></Provider>).toJSON();
    expect(DOM).toMatchSnapshot();
  });
});