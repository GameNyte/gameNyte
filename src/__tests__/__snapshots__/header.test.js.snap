// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Header renders correctly 1`] = `
<div
  className="makeStyles-header-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <h1
      className="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
    >
      Game Nyte
    </h1>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        Open Modal
      </button>
    </div>
  </header>
</div>
`;
